cmd/translatesfx/translatesfx/sa_test.go

```diff
@@ -132,53 +120,55 @@ func TestExpand_SAComplex(t *testing.T) {
 }
 
 func TestProcessDirective_Simple(t *testing.T) {
-	d := map[interface{}]interface{}{
+	d, err := parseDirective(map[interface{}]interface{}{
 		"#from": "testdata/token",
-	}
-	v, flatten, err := processDirective(d, "")
+	})
+	require.NoError(t, err)
+	v, err := processDirective(d, "")
 	require.NoError(t, err)
-	assert.False(t, flatten)
 	assert.Equal(t, "abc123", v.(string))
 }
 
 func TestProcessDirective_Map(t *testing.T) {
-	d := map[interface{}]interface{}{
+	d, err := parseDirective(map[interface{}]interface{}{
 		"#from": "testdata/map.yaml",
-	}
-	v, flatten, err := processDirective(d, "")
+	})
+	require.NoError(t, err)
+	v, err := processDirective(d, "")
 	require.NoError(t, err)
-	assert.False(t, flatten)
 	assert.Equal(t, map[interface{}]interface{}{
 		"foo": "bar",
 		"baz": "glarch",
 	}, v)
 }
 
 func TestProcessDirective_Wildcard(t *testing.T) {
-	d := map[interface{}]interface{}{
+	d, err := parseDirective(map[interface{}]interface{}{
 		"#from": "testdata/cfgs/map*.yaml",
-	}
-	v, flatten, err := processDirective(d, "")
+	})
+	require.NoError(t, err)
+	v, err := processDirective(d, "")
 	require.NoError(t, err)
-	assert.False(t, flatten)
 	assert.NotNil(t, v)
```

Comment:
not required but in general it could be helpful to have explicit return values even if repeated since the directive is fundamentally different

Reply:
Agreed. This gets reworked in a later PR though so I'll make sure this is addressed there.